provider/aws/aws.go

```diff
@@ -687,9 +761,28 @@ func (p *AWSProvider) newChange(action string, ep *endpoint.Endpoint) (*route53.
 		change.ResourceRecordSet.HealthCheckId = aws.String(prop.Value)
 	}
 
+	if ownedRecord, ok := ep.Labels[endpoint.OwnedRecordLabelKey]; ok {
```

Comment:
I get the point of that label. However, the `Labels` are intended to be persisted in the TXT records such that ExternalDNS can access that information across restarts or if multiple ExternalDNS' work within the same Hosted Zone.

However, I don't see that this persistence is needed and that this label is only required to hold this information while the Endpoint is processed within ExternalDNS during a single synchronisation run (comparable to `failedChangesQueue` which gets lost when ExternalDNS stops).

For this particular need (where persistence isn't needed) one should use the map `endpoint.ProviderSpecific` instead of `endpoint.Labels` because these are not persisted.

Reply:
These `Labels` are only added to the `TXT` records themselves; the labels of these are not persisted (it's the other way around, the TXT records are persisting the labels of the records they "own"). I would disagree that this particular field should be in `ProviderSpecific` instead as this ownership relation itself is not provider-specific in any way and could be picked up by the `ApplyChanges` implementation of any other provider, too.

Reply:
Ok, I see. If it's only on the TXT records and not persisted (because TXT records don't have TXT records for themselves to store them) then I don't mind.